script.js

```diff
@@ -1,4 +1,300 @@
+// The different game modes or turns
+var GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST = "CHOOSE HIGHEST_OR_LOWEST";
+var GAME_MODE_CHOOSE_NUMBER_OF_DICE = "CHOOSE_NUMBER_OF_DICE";
+var GAME_MODE_PLAYER_ONE_ROLL = "PLAYER_ONE_ROLL";
+var GAME_MODE_PLAYER_ONE_CHOOSE = "PLAYER_ONE_CHOOSE";
+var GAME_MODE_PLAYER_TWO_ROLL = "PLAYER_TWO_ROLL";
+var GAME_MODE_PLAYER_TWO_CHOOSE = "PLAYER_TWO_CHOOSE";
+
+// Capture user input
+var userInputs = [];
+var userInput = null;
+var gameMode = GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST;
+
+// Arrays and variables to capture user dice rolls and number results
+var player1DiceRolls = [];
+var player1Number = 0;
+var player1NumberList = [];
+var player1Score = 0;
+var player2DiceRolls = [];
+var player2Number = 0;
+var player2NumberList = [];
+var player2Score = 0;
+var numberOfDice = 0;
+var numberOfDices = [];
+var diceIndex = 0;
+var inputIndex = 0;
+var numberIndex = 0;
+var numberOfDiceIndex = 0;
+
+// Dice Roll function
+var rollDice = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var chooseHighestModeOrLowestMode = function () {
+  // Decide how the game will auto-generate combined number depending on the mode chosen.
+  // User input validation
+  if (userInput != "highest" && userInput != "lowest") {
+    return `Please enter either "highest" to play the 'highest combined number mode' or "lowest" to play the 'lowest combined number mode'.`;
+  }
+
+  userInputs.push(userInput);
+
+  if (userInputs[inputIndex] == "highest") {
+    var outputText = `highest`;
+  } else if ((userInputs[inputIndex] = "lowest")) {
```

Comment:
this could potentially be else instead of else if, since you have already validated the user input in front, anything that is not highest would already be lowest